src/Microsoft.AspNet.Razor/Generator/ExpressionCodeGenerator.cs

```diff
@@ -30,12 +32,13 @@ public override string ToString()
 
         public override bool Equals(object obj)
         {
-            return obj is ExpressionCodeGenerator;
+            return obj != null &&
```

Comment:
`HybridCodeGenerator` is `abstract` (for some reason) and has no other subclass.  so it really doesn't matter today.

my preference would be to delete `HybridCodeGenerator` but that doesn't fit in this PR at all. so I'll leave this as-is.